include/hip/hcc_detail/hip_runtime_api.h

```diff
@@ -1521,6 +1532,23 @@ hipError_t hipMemsetD8(hipDeviceptr_t dest, unsigned char value, size_t sizeByte
  */
 hipError_t hipMemsetAsync(void* dst, int value, size_t sizeBytes, hipStream_t stream __dparm(0));
 
+/**
+ *  @brief Fills the memory area pointed to by dev with the constant integer
+ * value for specified number of times.
+ *
+ *  hipMemsetD32Async() is asynchronous with respect to the host, so the call may return before the
+ * memset is complete. The operation can optionally be associated to a stream by passing a non-zero
+ * stream argument. If stream is non-zero, the operation may overlap with operations in other
+ * streams.
+ *
+ *  @param[out] dst Pointer to device memory
+ *  @param[in]  value - Value to set for each byte of specified memory
+ *  @param[in]  count - number of values to be set
+ *  @param[in]  stream - Stream identifier
+ *  @return #hipSuccess, #hipErrorInvalidValue, #hipErrorMemoryFree
+ */
+hipError_t hipMemsetD32Async(void* dst, int value, size_t count, hipStream_t stream __dparm(0));
```

Comment:
I believe this should be hipDeviceptr_t dst here.

Reply:
@mangupta could you help clarify when `void*` be used, and when `hipDeviceptr_t` be used? Looking at the header it seems they are used somewhat interchangeably.

Reply:
meanwhile I just pushed a commit to use `hipDeviceptr_t`

Reply:
Depends on what is the equivalent CUDA API signature. Typically CUDA runtime APIs use void* while CUDA driver APIs use CUdeviceptr* (which gets replaced by hipDeviceptr_t in the case of HIP APIs).